pkg/cronjob/es_rollover.go

```diff
@@ -213,6 +213,13 @@ func EsScriptEnvVars(opts v1.Options) []corev1.EnvVar {
 			envs = append(envs, corev1.EnvVar{Name: x.envVar, Value: val})
 		}
 	}
+
+	if val, ok := options["skip-dependencies"]; ok {
```

Comment:
Why do we need to add the environment var if the option is specified?

Reply:
For cronjobs, only selected parameters are passed as environment variables. Here we pass user settings if they exist. If no user settings are present and the operator runs on openshift, we set `skip-dependencies=true`.

Details: https://github.com/jaegertracing/jaeger-operator/issues/1896#issuecomment-1135894828